File: frontend/lib/models/supplier.dart
class Supplier {
  final int? supplierNumber;
  final String supplierName;
  final String? phone;
  final String? email;
  final String status;

  Supplier({
    this.supplierNumber,
    required this.supplierName,
    this.phone,
    this.email,
    required this.status,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) {
    return Supplier(
      supplierNumber: (json['supplierNumber'] as num?)?.toInt(),
      supplierName: json['supplierName'] ?? '',
      phone: json['phone'],
      email: json['email'],
      status: json['status'] ?? 'ACTIVE',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'supplierNumber': supplierNumber,
      'supplierName': supplierName,
      'phone': phone,
      'email': email,
      'status': status,
    };
  }
}

File: backend/src/main/java/com/supermarket/backend/entity/Supplier.java
package com.supermarket.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "suppliers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Supplier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "supplier_number")
    private Integer supplierNumber;

    @Column(name = "supplier_name")
    private String supplierName;

    @Column(name = "phone")
    private String phone;

    @Column(name = "email")
    private String email;

    @Column(name = "status")
    private String status;

    @Column(name = "contact_person")
    private String contactPerson;

    @Column(name = "address")
    private String address;

    @Column(name = "category")
    private String category;

    @Column(name = "notes")
    private String notes;
}
